lib/DxilPIXPasses/PixPassHelpers.cpp

```diff
@@ -450,6 +450,56 @@ void ReplaceAllUsesOfInstructionWithNewValueAndDeleteInstruction(
   delete Instr;
 }
 
+unsigned int FindOrAddSV_Position(hlsl::DxilModule &DM,
+                                  llvm::StringRef UpStream) {
+  hlsl::DxilSignature &InputSignature = DM.GetInputSignature();
+  auto &InputElements = InputSignature.GetElements();
+
+  auto Existing_SV_Position =
+      std::find_if(InputElements.begin(), InputElements.end(),
+                   [](const std::unique_ptr<DxilSignatureElement> &Element) {
+                     return Element->GetSemantic()->GetKind() ==
+                            hlsl::DXIL::SemanticKind::Position;
+                   });
+
+  // SV_Position, if present, has to have full mask, so we needn't worry
+  // about the shader having selected components that don't include x or y.
+  // If not present, we add it.
+  if (Existing_SV_Position == InputElements.end()) {
+
+    unsigned int Index = static_cast<unsigned int>(InputElements.size());
+
+    unsigned int StartRow = Index;
+    unsigned int StartColumn = 0;
+    unsigned int RowCount = 1;
+    unsigned int ColumnCount = 4;
+    if (!UpStream.empty()) {
+      SmallVector<StringRef, 4> indices;
+      UpStream.split(indices, "-");
+      if (indices.size() == 4) {
+        // These are encoded in ./DxilPixReadOutputSig.cpp
+        StartRow = atoi(indices[0].str().c_str());
+        StartColumn = atoi(indices[1].str().c_str());
+        RowCount = atoi(indices[2].str().c_str());
+        ColumnCount = atoi(indices[3].str().c_str());
```

Comment:
I think this is only valid if all of the following conditions are true, right?  Shouldn't we check for these and bail otherwise?
- `StartColumn == 0`
- `RowCount == 1`
- `ColumnCount == 4`

Reply:
Oh yeah, given that these all have to be these values, there's probably no value in passing anything but an int row index, similar to the prior option.

Reply:
Fair point. At first I thought it was better to rely entirely on the read-back-signature pass to get all these values precisely as they were in the upstream output sig. But if it's really true that these system values always start at col 0, then it's better to restrict the input to the one value (start row).